packages/vertexai/src/requests/schema-builder.ts

```diff
@@ -0,0 +1,197 @@
+import { SchemaInterface, SchemaType, SchemaParams } from '../types/schema';
+
+export abstract class Schema implements SchemaInterface {
+  /**
+   * Optional. The type of the property. {@link
+   * SchemaType}.
+   */
+  type: SchemaType;
+  /** Optional. The format of the property.
+   * Supported formats:
+   *  for NUMBER type: "float", "double"
+   *  for INTEGER type: "int32", "int64"
+   *  for STRING type: "email", "byte", etc
+   */
+  format?: string;
+  /** Optional. The description of the property. */
+  description?: string;
+  /** Optional. Whether the property is nullable. Defaults to false. */
+  nullable: boolean;
+  /** Optional. Defaults to true. */
+  required: boolean;
+  /** Optional. The example of the property. */
+  example?: unknown;
+
+  constructor(schemaParams: SchemaInterface) {
+    this.type = schemaParams.type;
+    this.format = schemaParams?.format;
+    this.description = schemaParams?.description;
+    this.example = schemaParams?.example;
+    this.nullable = schemaParams.hasOwnProperty('nullable')
+      ? !!schemaParams.nullable
+      : false;
+    this.required = schemaParams.hasOwnProperty('required')
+      ? !!schemaParams.required
+      : true;
+  }
+
+  /** Converts class to a plain JSON object (not a string). */
+  toJSON(): Record<string, unknown> {
```

Comment:
I'm not sure what the best name is - this is an intermediate object between the very abstract Schema, where all the children are Schemas, and a JSON string - it's a plain JS object or POJO, could call it `toPOJO()`? I have this instead of going straight to the stringified JSON because it's easier to inspect and debug.

Reply:
Oh, I was wrong, toJSON() actually should not return a JSON string, it should in fact return a JS symbol that is ready to pass to JSON.stringify(). Changing this back. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#tojson_behavior